server/verify.go

```diff
@@ -175,44 +187,44 @@ func VerifyAttestation(attestation *pb.Attestation, opts VerifyOpts) (*pb.Machin
 
 // validateAK validates AK cert in the attestation, and returns AK cert (if exists) and public key.
 // It also pulls out the GCE Instance Info if it exists.
-func validateAK(attestation *pb.Attestation, opts VerifyOpts) (*pb.MachineState, crypto.PublicKey, error) {
+func validateAK(attestation *pb.Attestation, opts VerifyOpts) (*pb.MachineState, tpm2.Public, crypto.PublicKey, error) {
+	// If the AK Cert is not in the attestation, use the AK Public Area.
```

Comment:
akpub was optional if akcert presents (even though we attach it anyway), so I don't think we should return error if it cannot get akpub

instead, we can return error when it specifically require HashNonce
```
if err != nil and opts.HashNonce {
    // error
}
```

Reply:
I've opted to not include this check (for a breaking change) since it would propogate the `if opts.HashNonce` check down to every new new check in this func (decoding, getting crypto key, comparing to cert key) for questionable value.

Reply:
we may want to consider remove the optional: https://github.com/google/go-tpm-tools/blob/d2379fa8b41901accaa7017e56731dac97d02987/proto/attest.proto#L35